src/dds/proc.nix.cpp

```diff
@@ -67,6 +67,15 @@ ::pid_t spawn_child(const proc_options& opts, int stdout_pipe, int close_me) noe
     std::_Exit(-1);
 }
 
+struct fd_closer {
```

Comment:
I had a quick look bu couldn't see a `scope_guard` type in use in the project so I made a light RAII wrapper - if I've missed the better way to achieve this I'm happy to change it! :)

Reply:
There's a "defer" macro in `<neo/scope.hpp>`, which would look like:

```c++
neo_defer { ::close(read_pipe); };
```